java/src/main/java/api/example/java/controllers/AgronomicDataController.java

```diff
@@ -0,0 +1,89 @@
+package api.example.java.controllers;
+
+import java.io.IOException;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import api.example.java.api.ClimateAPIs;
+import api.example.java.model.Activity;
+import api.example.java.model.ActivityResult;
+
+@Controller
+public class AgronomicDataController extends BaseController {
+	
+	
+	@Autowired
+	private ClimateAPIs climateAPIs;
+
+	private static Logger logger = LoggerFactory.getLogger(AgronomicDataController.class);
+
+	@GetMapping("/agronomic")
+	public String agronomicData(Model model, @RequestParam("data") String dataType, HttpServletRequest request) {
+		
+		logger.info("Listing agronic {} data"+ dataType);
+
+		List<Activity> acitivities = new ArrayList<Activity>();
+
+		Iterator<ActivityResult> acticityIterator = climateAPIs.getActivityIterator(
+				String.format("%s/%s", agronomicApiUri(), dataType)
+				, dataType, 
```

Comment:
Minor - move comma to the previous line?

Reply:
fixed